geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java

```diff
@@ -56,33 +52,31 @@ public RedisKey(byte[] value) {
       endHashtag = value.length;
     }
 
-    // & (REDIS_SLOTS - 1) is equivalent to % REDIS_SLOTS but supposedly faster
-    routingId = (CRC16.calculate(value, startHashtag + 1, endHashtag) & (REDIS_SLOTS - 1))
-        / REDIS_SLOTS_PER_BUCKET;
+    crc16 = CRC16.calculate(value, startHashtag + 1, endHashtag);
   }
 
   /**
    * Used by the {@link RedisPartitionResolver} to map slots to buckets
    */
-  public Object getRoutingId() {
-    return routingId;
+  public Integer getCrc16() {
+    return crc16;
   }
 
   @Override
   public int getDSFID() {
-    return DataSerializableFixedID.REDIS_BYTE_ARRAY_WRAPPER_KEY;
+    return DataSerializableFixedID.REDIS_KEY;
   }
 
   @Override
   public void toData(DataOutput out, SerializationContext context) throws IOException {
-    DataSerializer.writeInteger(routingId, out);
+    out.writeInt(crc16);
     super.toData(out, context);
   }
 
   @Override
   public void fromData(DataInput in, DeserializationContext context)
       throws IOException, ClassNotFoundException {
-    routingId = DataSerializer.readInteger(in);
+    crc16 = in.readInt();
```

Comment:
Wondering out loud now if it is even worth serializing the CRC when we could just reconstitute it when deserializing.